drafts/client-interface.md

```diff
@@ -3,11 +3,150 @@
 This document describes the interface of a yet-to-be-built reference client
 library in pseudo code.
 
+The hydra client works similar to a browser. You point it to a URL. The URL becomes the current resource. 
+Then you interact with the controls of the resource.
 
-## Retrieve a resource
+The examples show how a cli application could use the hydra client for interaction with an API.
+
+## Find API entrypoint for website
+
+Hydra has a special mechanism to discover API entrypoints.
+
+```
+hydra.findEntrypoint(website)
+print hydra.representation 
+// the API entrypoint is now the current representation
```

Comment:
I had started like that and found the resulting code difficult to handle. We could certainly rewrite hydra to return the representation, but let us consider a bit first.

The question is, should the application that uses the hydra client maintain the application state or should the hydra client do that?

"Not easy to understand" might indicate that it is a novel approach. Current http clients "retrieve stuff". The hydra client is different. It is an application, and you drive it by interacting with it.